test/sql/test_case_statement.py

```diff
@@ -294,3 +294,20 @@ def testcase_with_dict(self):
             ("two", 2),
             ("other", 3),
         ]
+
+    def test_type_of_case_expression_with_null_case_and_no_else_clause(self):
+        expr = case(
+            (info_table.c.pk < 0, None),
+            (info_table.c.pk <= 9, info_table.c.pk),
+            (info_table.c.pk > 9, None),
```

Comment:
**Federico Caselli** (CaselIT) wrote:

let's add 2 whens with different types here so we verify the resolution order too

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5084